server/selective_sync.go

```diff
@@ -35,9 +51,76 @@ func (p *Plugin) ChatMembersSpanPlatforms(members []*model.ChannelMember) (bool,
 
 		if p.IsRemoteUser(user) {
 			// Synthetic users are always remote.
+			atLeastOneRemoteUser = true
+		} else if p.getPrimaryPlatform(user.Id) == storemodels.PreferenceValuePlatformMSTeams {
+			// Treat Teams primary users as remote
+			atLeastOneRemoteUser = true
+		} else {
+			// Otherwise the user is considered local.
+			atLeastOneLocalUser = true
+		}
+
+		if atLeastOneLocalUser && atLeastOneRemoteUser {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func (p *Plugin) ChannelConnectedOrRemote(channelID, senderID string) (bool, error) {
+	senderConnected, err := p.IsUserConnected(senderID)
+	if err != nil {
+		// ah.plugin.GetAPI().LogWarn("Unable to determine if sender is connected", "error", err.Error())
```

Comment:
Should we keep this log?

Reply:
It gets logged in the caller `handleCreatedActivity`, so no not needed, but I did remove the commented out line.